### DISCUSSION TITLE
Adding CMAKE_SYSTEM_PROCESSOR to output directory names

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#2323] [Ideas]
We are building our deployment on CI for both x64 Linux and aarch64 Linux, but we are finding that because both binaries are output to `build-artifacts/Linux/Topology/bin/Topology`, the binary built second overwrites the one built first.

Is anyone else having this problem?

Would anyone be against changing the output directory to `build-artifacts/${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}`?

**Comment by LeStarch** [comment]
Seems like a fine recommendation!

Are you using a custom toolchain for `aarch64` or are you compiling on an `aarch64` host?  Our aarch64 toolchains set CMAKE_SYSTEM_NAME to be `aarch64-linux` and thus we haven't seen the collision.  I am curious about your configuration as I wonder if other things might need minor adjusting too!

**Reply by JackNWhite** [reply]
We have a manufacturer cross-compiler which comes with scripts to set a load of environment variables, e.g. CC, CXX, LD. It also supplies a CMake toolchain script which sets `CMAKE_SYSTEM_NAME` to `Linux` and `CMAKE_SYSTEM_PROCESSOR` to `aarch64`, which I believe is what CMake prescribes.

The F' aarch64 toolchain (_Fprime/cmake/toolchain/aarch64-linux.cmake_) sets these variables to the same thing, and if building without an explicit toolchain, to `Linux` and `x86_64`.

**Reply by LeStarch** [reply]
Thanks for the info!  I agree we should track this as an improvement.  We'll convert it to an issue.
